int32 number_of_walls
float64[] points
